### PR TITLE
Uncommented existing test_cli for "oaei-ordo-hp"

### PR DESCRIPTION
**Pull request by hrshdhgd**
The test for parsing `oaei-ordo-hp.rdf` was commented. This PR just uncomments it and wires is properly in `test_cli.py` to verify if it passes.

### CODE REVIEW

**Comment by hrshdhgd** [comment]
@matentzn : ok, so I'm running :```sssom parse tests/data/oaei-ordo-hp.rdf -o test.tsv --input-format alignment-api-xml ```

and the test.tsv begins
```
# curie_map:
#   HP: http://purl.obolibrary.org/obo/HP_
#   orphanet.ordo: http://www.orpha.net/ORDO/Orphanet_
#   owl: http://www.w3.org/2002/07/owl#
#   rdf: http://www.w3.org/1999/02/22-rdf-syntax-ns#
#   rdfs: http://www.w3.org/2000/01/rdf-schema#
#   semapv: https://w3id.org/semapv/
#   skos: http://www.w3.org/2004/02/skos/core#
#   sssom: https://w3id.org/sssom/
# license: https://w3id.org/sssom/license/unspecified
# mapping_set_id: https://w3id.org/sssom/mappings/105b111d-1f45-44e3-bb6d-8c33670537b7
# object_source: http://purl.obolibrary.org/hp.owl
# subject_source: http://purl.obolibrary.org/ordo.owl
subject_id	predicate_id	object_id	mapping_justification	confidence
orphanet.ordo:100047	owl:equivalentClass	HP:0100681	semapv:UnspecifiedMatching	0.27449
orphanet.ordo:100069	owl:equivalentClass	HP:0030219	semapv:UnspecifiedMatching	0.875
orphanet.ordo:100088	owl:equivalentClass	HP:0002890	semapv:UnspecifiedMatching	0.875
```

Is the `object_source` value incorrect? The reason I ask is `test_conversion.py` is throwing an error stating that the value is incorrect.

File: sssom/parsers.py [hunk 1]
```diff
@@ -540,8 +547,14 @@ def from_alignment_minidom(
                 elif node_name == "onto2":
                     ms[OBJECT_SOURCE_ID] = e.firstChild.nodeValue
                 elif node_name == "uri1":
-                    ms[SUBJECT_SOURCE] = e.firstChild.nodeValue
+                    # import pdb; pdb.set_trace()
+                    ms[
+                        SUBJECT_SOURCE
+                    ] = (
+                        e.firstChild.nodeValue
+                    )  # CURIEfy this node value && If endswith(".extension"), replace it with _)
```
Comment (matentzn) [inline sssom/parsers.py:555]: What is your plan here re curification if an URI prefix is not in the prefix map?

**Review by matentzn — COMMENTED** [review]
What is missing in this PR?

**Comment by hrshdhgd** [comment]
> What is missing in this PR?

Just the curieficaion

File: sssom/parsers.py (hunk 1 above)
Comment (matentzn) [reply sssom/parsers.py:555]: I would vote for leaving the URIs as is if the URI prefix is not known to the prefix map. We should produce a single warning at the end of the pipeline which lists the first 5 URIs that didn't have a proper CURIE, and log a warning "the extracted file is not valid SSSOM because some of the URIs could not be converted. Please provide appropriate URI prefixes and repeat  the process".


Charlie prefers to not process the node at all if the URI prefix is unknown; this is an alternative, but only if we can overwrite it with a flag, and then we are back to the first point.

Comment (hrshdhgd) [reply sssom/parsers.py:555]: That is the problem. The `sssom` definition for `subject_source` and `object_source` is `EntityReference`. Does this mean only CURIEs? `uriOrCurie` might be a better solution?

Comment (matentzn) [reply sssom/parsers.py:555]: I think a URI _is_ a curie, where the prefix is `http` and the URI expansion `http` so everwhere where there is an entity reference, you should be able to drop a URI.

Its fine to write a URI, just also write the warning I mention above.

File: sssom/util.py [hunk 2]
```diff
@@ -1440,7 +1440,7 @@ def get_all_prefixes(msdf: MappingSetDataFrame) -> list:
                         # print(
                         #     f"Slot '{slot}' has an incorrect value: {msdf.metadata[slot]}"
                         # )
-                        raise ValidationError(
+                        logging.warning(
```
Comment (matentzn) [inline sssom/util.py:1443]: Why was this necessary?

File: sssom/validators.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def check_all_prefixes_in_curie_map(msdf: MappingSetDataFrame) -> None:
 
     missing_prefixes = []
     for pref in prefixes:
-        if pref not in list(msdf.prefix_map.keys()):
+        if pref not in list(msdf.prefix_map.keys()) and pref != "":
```
Comment (matentzn) [inline sssom/validators.py:86]: Shouldn't pref != "" be tested first?

**Review by matentzn — COMMENTED** [review]
Looks great, two minor comments!

File: sssom/util.py (hunk 2 above)
Comment (matentzn) [reply sssom/util.py:1443]: This does not seem the right way to solve this though..  Which process adds an `object_source` that is not a valid CURIE?

Comment (hrshdhgd) [reply sssom/util.py:1443]: [test_conversion.py](https://github.com/mapping-commons/sssom-py/blob/h2-patch3/tests/test_conversion.py#L106) <= [writers.py](https://github.com/mapping-commons/sssom-py/blob/h2-patch3/sssom/writers.py#L107) <= [validators.py](https://github.com/mapping-commons/sssom-py/blob/h2-patch3/sssom/validators.py#L73)

Comment (matentzn) [reply sssom/util.py:1443]: None of this _adds_ an invalid object_source though that would justify this change in this PR now. 

Some observations (partially related):

1. OBJECT_SOURCE_ID does _not_ exist in SSSOM: https://github.com/mapping-commons/sssom-py/blob/master/sssom/parsers.py#L541
2. The URI parameter should validate correctly as an object_source: https://github.com/mapping-commons/sssom-py/blob/master/sssom/parsers.py#L543

Can you say what exactly the error is saying?

Comment (hrshdhgd) [reply sssom/util.py:1443]: https://github.com/mapping-commons/sssom-py/actions/runs/5137740141/jobs/9246227909

Comment (matentzn) [reply sssom/util.py:1443]: Got it! Proceed then, its not the "validate" methods, its something else. Thanks!
